### ISSUE TITLE
Update not triggered

### ISSUE THREAD

**Issue by swensson** [betula/use-between#25]
Hey there! I'm making a simple testing scenario and I can't get my `Header` component (code below):

```
export default () => {
  const { user, loading } = useBetween(useUser);

  console.log('header', {user, loading});

  /**
   * Navigation callbacks for the right part of the header
   */
  const goToDashboard = React.useCallback(() => history.push('/dashboard'), []);
  const goToLogin = React.useCallback(() => history.push('/'), []);
  const goToProfile = React.useCallback(() => history.push('/profile'), []);

  return (
  // ....
```

to grab updates from `useUser`, following code:

```
const useUser = () => {
  const [user, setUser] = React.useState(null as any);
  const [loading, setLoading] = React.useState(true);

  const fetchUser = React.useCallback(() => {
    console.log('fetch user');
    backend.session.get().then((user) => {
      console.log('set user', user);
      setUser(user);
    }).catch(() => {
      // nope here, just keep user null meaning
      // we are not logged in
    }).then(() => {
      setLoading(false);
    })
  }, [setUser, setLoading])

  React.useEffect(() => {
    fetchUser();
  }, []);

  console.log('user', user);

  return { user, loading };
};
```

here's what my console output looks like on startup:

```
user.ts:24 fetch user
index.tsx:58 header {user: null, loading: true}
user.ts:26 set user {id: 1, email: 'qwe@qwe.qwe', name: 'Alex Swensson', role: 'admin', bio: '# Me\n\n...', …}
user.ts:38 user {id: 1, email: 'qwe@qwe.qwe', name: 'Alex Swensson', role: 'admin', bio: '# Me\n\n...', …}
user.ts:38 user {id: 1, email: 'qwe@qwe.qwe', name: 'Alex Swensson', role: 'admin', bio: '# Me\n\n...', …}
index.js:573 [webpack-dev-server] Hot Module Replacement enabled.
index.js:573 [webpack-dev-server] Live Reloading enabled.
```

as you can see the update in `useUser` is triggered, but it didn't appear in `Header` components. What can I be possibly doing wrong?

**Comment by swensson** [comment]
Turns out wrapping fetchUser in a timeout works fine. What can this be related to?
```
React.useEffect(() => {
    setTimeout(() => {
      fetchUser();
    }, 0);
  }, []);
```

**Comment by betula** [comment]
Hi @swensson,

Great catch! Thanks! I took a look and will fix it.

**Comment by betula** [comment]
Hi @swensson,

I have great news! I fixed it in version [1.3.1](https://github.com/betula/use-between/releases/tag/v1.3.1)

Please, Try it Out!
and Welcome to discussion!

**Comment by swensson** [comment]
I can see this one works great now! Thank you @betula
